lib/middleware.js

```diff
@@ -1,137 +1,93 @@
-  const manifest = readJsonSync(manifestName, options.root);
-  if (manifest !== null) {
-    return manifest.name;
-  }
-  const basename = path.basename(options.root || process.cwd());
-  console.warn(
-      `no ${manifestName} found, defaulting to packageName=${basename}`);
-  return basename;
+  return cache[asset];
 }
 
 /**
  * Middleware that serves an instrumented asset based on user
  * configuration of coverage
  */
 function coverageMiddleware(root, options, emitter) {
-  const basename = getPackageName(options);
-  const basepath = emitter.options.clientOptions.root + basename + '/';
+  const basename = util.getPackageName(options);
+  const basepath = path.join(emitter.options.clientOptions.root, basename);
 
-  return function(req, res, next) {
-    var absolutePath = req.url.replace(/^\/[^/]+\/[^/]+/, root);
+  return function (req, res, next) {
+    let blacklist = options.exclude || ['**/test/**'];
+    let whitelist = options.include || [];
 
-    // always ignore platform files in addition to user's blacklist
-    var blacklist = ['/web-component-tester/*'];
-    if(options.exclude) {
-      blacklist = blacklist.concat(options.exclude);
-    } else {
-      blacklist = blacklist.concat(['**/test/**']);
+    if (!options.ignoreBasePath) {
+      blacklist = blacklist.map(x => path.join(basepath, x));
+      whitelist = whitelist.map(x => path.join(basepath, x));
     }
-    var whitelist = options.include.map(x => basepath + x);
 
-    // cache the webserver root for user supplied instrumenter
-    this.root = root;
-    // check asset against rules
-    var process = match(req.url, whitelist) && !match(req.url, blacklist);
-
-    // instrument unfiltered assets
-    if ( process ) {
+    if (match(req.url, whitelist) && !match(req.url, blacklist)) {
       emitter.emit('log:debug', 'coverage', 'instrument', req.url);
 
-      if (absolutePath.match(/\.htm(l)?$/)) {
-        var html = instrumentHtml(absolutePath, req);
-        return res.send( html );
+      const re = new RegExp(`^\/[^/]+\/${basename.replace('/', '\/')}`);
+      const absolutePath = req.url.replace(re, root);
+
+      if (absolutePath.match(/\.js?$/)) {
+        let html = instrumentFile(absolutePath, req);
+        res.type('application/javascript');
+        html = html.replace('coverage = global[gcv] || (global[gcv] = {});', 'coverage = global.WCT.share.__coverage__ || (global.WCT = { share: { __coverage__: {} } }); ');
+        return res.send(transform(req, html, basename, absolutePath, options.npm, root, emitter.options.clientOptions.root));
+      } else if (absolutePath.match(/\.htm(l)?$/)) {
+        let html = instrumentFile(absolutePath, req, true);
+        return res.send(html.replace('coverage = global[gcv] || (global[gcv] = {});', 'coverage = global.WCT.share.__coverage__ || (global.WCT = { share: { __coverage__: {} } }); '));
       }
 
-      return res.send( instrumentAsset(absolutePath, req) );
+      return next();
```

Comment:
Technically the instrument debug log is still emitted, even though the code has not been instrumented. Maybe make a distinction for that?